### PR TITLE
[mildom] Rework extractors

### PR DESCRIPTION
**Pull request by Lesmiscore**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Reworked Mildom extractors.

- Fixed to request token for HLS
- Removed all requests needed to generate common queries. Now all of them are redundant and no longer initialization.
- Added support for clips.

### CODE REVIEW

File: yt_dlp/extractor/mildom.py [hunk 1]
```diff
@@ -287,29 +274,28 @@ class MildomUserVodIE(MildomBaseIE):
             'id': '10093333',
             'title': 'Uploads from ねこばたけ',
         },
-        'playlist_mincount': 351,
+        'playlist_mincount': 732,
```
Comment (pukkandan) [inline yt_dlp/extractor/mildom.py:277]: any point in increasing the mincount?

File: yt_dlp/extractor/mildom.py [hunk 2]
```diff
@@ -287,29 +274,28 @@ class MildomUserVodIE(MildomBaseIE):
...
+        reply = self._call_api(
+            'https://cloudac.mildom.com/nonolive/videocontent/profile/playbackList',
+            user_id, note=f'Downloading page {page}', query={
+                'user_id': user_id,
+                'page': page,
+                'limit': '30',
+            })
+        if not reply:
+            return
+        yield from (self.url_result(f'https://www.mildom.com/playback/{user_id}/{x["v_id"]}') for x in reply)
```
Comment (pukkandan) [inline yt_dlp/extractor/mildom.py:298]:
```suggestion
        for x in reply:
            yield self.url_result(f'https://www.mildom.com/playback/{user_id}/{x["v_id"]}')
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/mildom.py (hunk 1 above)
Comment (Lesmiscore) [reply yt_dlp/extractor/mildom.py:277]: To notice bugs in paging in the future (or massive deletion)

File: yt_dlp/extractor/mildom.py [hunk 3]
```diff
@@ -1,104 +1,49 @@
...
 from ..utils import (
-    std_headers,
-    update_url_query,
-    random_uuidv4,
-    try_get,
+    ExtractorError,
+    OnDemandPagedList,
+    determine_ext,
+    dict_get,
     float_or_none,
```
Comment (Ashish0804) [inline yt_dlp/extractor/mildom.py:13]: sort

File: yt_dlp/extractor/mildom.py [hunk 4]
```diff
@@ -216,9 +146,8 @@ class MildomVodIE(MildomBaseIE):
     }]
 
     def _real_extract(self, url):
-        m = self._match_valid_url(url)
-        user_id, video_id = m.group('user_id'), m.group('id')
-        url = 'https://www.mildom.com/playback/%s/%s' % (user_id, video_id)
+        user_id, video_id = self._match_valid_url(url).group('user_id', 'id')
+        url = f'https://www.mildom.com/playback/{user_id}/{video_id}'
```
Comment (Ashish0804) [inline yt_dlp/extractor/mildom.py:150]: why make this a var?
Don't see it being used anywhere other than webpage call...

File: yt_dlp/extractor/mildom.py [hunk 5]
```diff
@@ -266,12 +181,80 @@ def _real_extract(self, url):
...
+        webpage = self._download_webpage(f'https://www.mildom.com/clip/{video_id}', video_id)
+
+        clip_detail = self._call_api(
+            'https://cloudac-cf-jp.mildom.com/nonolive/videocontent/clip/detail', video_id,
+            note='Downloading playback metadata', query={
+                'clip_id': video_id,
+            })
+
+        formats = [{
+            'url': clip_detail.get('url'),
```
Comment (Ashish0804) [inline yt_dlp/extractor/mildom.py:245]: i personally would have made this fatal...
It's just 1 url, why even set in format?
Just return it in the `url` keyword

File: yt_dlp/extractor/mildom.py [hunk 6]
```diff
@@ -287,29 +270,29 @@ class MildomUserVodIE(MildomBaseIE):
...
+            'https://cloudac.mildom.com/nonolive/videocontent/profile/playbackList',
+            user_id, note=f'Downloading page {page}', query={
+                'user_id': user_id,
+                'page': page,
+                'limit': '30',
+            })
+        if not reply:
+            return
+        for x in reply:
+            yield self.url_result(f'https://www.mildom.com/playback/{user_id}/{x["v_id"]}')
```
Comment (Ashish0804) [inline yt_dlp/extractor/mildom.py:295]:
```suggestion
        for x in reply:
            v_id = x.get('v_id')
            if not v_id:
                continue
            yield self.url_result(f'https://www.mildom.com/playback/{user_id}/{v_id}')
```

File: yt_dlp/extractor/mildom.py (hunk 4 above)
Comment (Lesmiscore) [reply yt_dlp/extractor/mildom.py:150]: didn't notice until you wrote about it here

**Pull request merged**